### PR TITLE
Add `--diff` flag to `nf-core modules update`

### PR DESCRIPTION
**Pull request by ErikDanielsson**
Added new `--diff` flag to `nf-core modules update`. When the flag is supplied the diff between the installed files and the new files is shown, and the user is asked to confirm the installation.

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [x] `CHANGELOG.md` is updated
 - [ ] If you've fixed a bug or added code that should be tested, add tests!
 - [x] Documentation in `docs` is updated

### CODE REVIEW

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Very nice! 🥇 

I tried to use it to check the difference of the `rseqc/bamstat` module since it was recently updated on this [PR](https://github.com/nf-core/modules/pull/654). The version on the rnaseq pipeline has not yet this changes I run within this pipeline the following command:

```console
$ nf-core modules update rseqc/bamstat --diff

                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 2.2.dev0



INFO     'nf-core/modules/rseqc/bamstat' is already up to date
```

However the command says that the module is already up to date. Actually, I also check the command without the `--diff` flag and get the same result. So might be that this is indeed something failing on the `update` command.

The change in the module is on this [line](https://github.com/nf-core/modules/blob/28b023e6f4d0d2745406d9dc6e38006882804e67/modules/rseqc/bamstat/main.nf#L14) and [here](https://github.com/nf-core/rnaseq/blob/064994beb278056d1439857c173a4b24955e5d74/modules/nf-core/modules/rseqc/bamstat/main.nf#L14) is the current version of the module installed on the rnaseq pipeline

**Comment by ErikDanielsson** [comment]
Yes, that probably indicates that the bug is in the `update` command rather than the new code. I'll have a look, thanks for finding it!

**Comment by ErikDanielsson** [comment]
Aha, the issue is that `rseqc/bamstat` has an entry in the `.nf-core.yml` in `rnaseq` @JoseEspinosa! I'll add a log message about that though.

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@
 ### Modules
 
 * Fixed typo in `module_utils.py`.
+* Added `--diff` flag to `nf-core modules update` which shows the diff between the installed files and the versions
```
[the remark this answers is not in the document]
Comment (ErikDanielsson) [reply CHANGELOG.md:14]: Yeah I thought it was more apt when I added the option of installing the files anyways.

**Comment by JoseEspinosa** [comment]
> Aha, the issue is that `rseqc/bamstat` has an entry in the `.nf-core.yml` in `rnaseq` @JoseEspinosa! I'll add a log message about that though.

Aaarg!!! Sorry I should have check it! Yes that will be better for people like me 😓

**Review by JoseEspinosa — APPROVED** [review]
LGTM! 🚀

**Comment by ErikDanielsson** [comment]
Thanks for the review @JoseEspinosa!

**Pull request merged**